### PR TITLE
fix(lsp): auto-discover deno.json in more cases

### PR DESCRIPTION
**Pull request by dsherret**
We weren't auto-discovering the deno.json in two cases:

1. A project that didn't have a deno.json and just added one.
2. After a syntax error in the deno.json.

This now rediscovers it in both these cases.

Closes https://github.com/denoland/vscode_deno/issues/867

### CODE REVIEW

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -1493,22 +1493,30 @@ impl Inner {
...
-          self.client.show_message(MessageType::WARNING, err);
-        }
-        touched = true;
+    let has_config_changed = match self.config.maybe_config_file() {
+      Some(config_file) => changes.contains(&config_file.specifier),
+      None => {
+        // check for auto-discovery
+        changes.iter().any(|url| {
+          url.path().ends_with("/deno.json")
+            || url.path().ends_with("/deno.jsonc")
```
Comment (bartlomieju) [inline cli/lsp/language_server.rs:1502]: Won't the slash cause problems on Windows?

File: cli/lsp/language_server.rs [hunk 2]
```diff
@@ -1493,22 +1493,30 @@ impl Inner {
...
+        })
       }
+    } || match self.config.maybe_lockfile() {
+      Some(lockfile) => has_lockfile_changed(&lockfile.lock(), &changes),
+      None => {
+        // check for auto-discovery
+        changes.iter().any(|url| url.path().ends_with("/deno.lock"))
```
Comment (bartlomieju) [inline cli/lsp/language_server.rs:1509]: Ditto

**Review by bartlomieju — APPROVED** [review]
LGTM, but I have a question

File: cli/lsp/language_server.rs (hunk 1 above)
Comment (dsherret) [reply cli/lsp/language_server.rs:1502]: They're urls :)

**Pull request merged**